src/math.rs

```diff
@@ -46,6 +47,41 @@ where
     clamped
 }
 
+#[derive(Debug, Clone, Copy)]
+pub struct ClosedInterval<T> {
+    pub lower_bound: T,
+    pub upper_bound: T,
+}
+
+impl<T> ClosedInterval<T>
+where
+    T: PartialOrd,
+{
+    pub fn contains(self, value: T) -> bool {
+        self.lower_bound <= value && value <= self.upper_bound
+    }
+}
+
+impl<T> FromStr for ClosedInterval<T>
+where
+    T: std::str::FromStr,
+{
+    type Err = T::Err;
+
+    fn from_str(s: &str) -> std::result::Result<Self, Self::Err> {
+        let bounds: Vec<&str> = s.split(',').collect();
+        assert_eq!(
```

Comment:
It would be nicer if a function with a Result couldn't panic – maybe this should have `Err = String` instead?

Reply:
It would be nicer indeed, but I didn't find an easy way to implement it...

Reply:
Can I merge in any case?

Reply:
Sure. But as for how to implement it with Err = String, you could of course do something like 
```rust
let mut parts = s.split(',');
let lower_bound = parts.next().ok_or("Empty string provided".into());
let upper_bound = parts.next().ok_or("Interval needs to have two comma-separated values".into());
if let Some(_) = parts.next() {
    return Err("Interval may not have more than two comma-separated values".into())
}
```

Reply:
I implemented an ParseClosedIntervalError, which is a little cleaner overall...